src/index.ts

```diff
@@ -125,83 +139,99 @@ export function useArray<T = any>(initial: T[]): UseArray<T> {
       }),
     [],
   )
-  return useMemo(
+  const actions = useMemo(
     () => ({
-      value,
       setValue,
       add,
       move,
       clear,
       removeById,
       removeIndex,
     }),
-    [add, clear, move, removeById, removeIndex, value],
+    [add, clear, move, removeById, removeIndex],
   )
+  return [value, actions]
```

Comment:
@kitze what do you think, does it makes sense to wrap this thing to `useMemo` too so if user not destructure it right away, but wants to pass it down the context, for example, to keep reference equality on each rerender?
E.g.
```jsx
const useAmazingArray = useArray([])
<Context.Provider value={{
   useAmazingArray
}}>
  {children}
</Context.Provider>
```
Maybe a bit of an overkill tho :)